packages/apollo-server-hapi/src/ApolloServer.ts

```diff
@@ -90,66 +93,97 @@ export class ApolloServer extends ApolloServerBase {
+                  (request.payload as any)
+                : request.query,
+            request: convertNodeHttpToRequest(request.raw.req),
+          },
+        );
+
+        const response = h.response(graphqlResponse);
+        if (responseInit.headers) {
+          Object.entries(responseInit.headers).forEach(([headerName, value]) =>
+            response.header(headerName, value),
           );
+        }
+        response.code(responseInit.status || 200);
+        return response;
+      } catch (error) {
+        if (!isHttpQueryError(error)) {
+          throw Boom.boomify(error as Error);
+        }
 
-          const response = h.response(graphqlResponse);
-          if (responseInit.headers) {
-            Object.entries(responseInit.headers).forEach(
-              ([headerName, value]) => response.header(headerName, value),
-            );
-          }
-          response.code(responseInit.status || 200);
-          return response;
-        } catch (error) {
-          if (!isHttpQueryError(error)) {
-            throw Boom.boomify(error as Error);
-          }
-
-          if (true === error.isGraphQLError) {
-            const response = h.response(error.message);
-            if (error.headers) {
-              Object.entries(error.headers).forEach(([headerName, value]) => {
-                response.header(headerName, value);
-              });
-            }
-            response.code(error.statusCode);
-            response.type('application/json');
-            return response;
-          }
-
-          const err = new Boom.Boom(error.message, {
-            statusCode: error.statusCode,
-          });
+        if (true === error.isGraphQLError) {
+          const response = h.response(error.message);
           if (error.headers) {
             Object.entries(error.headers).forEach(([headerName, value]) => {
-              err.output.headers[headerName] = value;
+              response.header(headerName, value);
             });
           }
-          // Boom hides the error when status code is 500
-          err.output.payload.message = error.message;
-          throw err;
+          response.code(error.statusCode);
+          response.type('application/json');
+          return response;
         }
-      },
+
+        const err = new Boom.Boom(error.message, {
+          statusCode: error.statusCode,
+        });
+        if (error.headers) {
+          Object.entries(error.headers).forEach(([headerName, value]) => {
+            err.output.headers[headerName] = value;
+          });
+        }
+        // Boom hides the error when status code is 500
+        err.output.payload.message = error.message;
+        throw err;
+      }
+    };
+
+    // POST ROUTE
+    const postOptions = { ...route, ...routePost };
+
+    // if we have post route options which contain no cors options, but have
+    // specified cors options separately, then merge them in
+    if (!!cors && postOptions.cors == null) {
```

Comment:
`!!` not needed in a boolean context like this.

`== null` means "null or undefined"; is that intentional?

Reply:
Intentional, because it's not only a boolean context. cors may be true / false or an object for configuration (see: https://hapi.dev/api?v=20.2.1#-routeoptionscors)